stream-chat-android-client/src/main/java/io/getstream/chat/android/client/uploader/StreamFileUploader.kt

```diff
@@ -0,0 +1,122 @@
+package io.getstream.chat.android.client.uploader
+
+import io.getstream.chat.android.client.api.RetrofitCdnApi
+import io.getstream.chat.android.client.api.models.ProgressRequestBody
+import io.getstream.chat.android.client.api.models.RetroProgressCallback
+import io.getstream.chat.android.client.extensions.getMediaType
+import io.getstream.chat.android.client.utils.ProgressCallback
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.asRequestBody
+import java.io.File
+
+internal class StreamFileUploader(
+    private val apiKey: String,
+    private val retrofitCdnApi: RetrofitCdnApi
+) : FileUploader {
+
+    private var userId: String = ""
+    private var connectionId: String = ""
```

Comment:
We could make these nullable, and throw errors if the methods of `FileUploader` are called before `setConnection` sets these values. Would be more explicit than something quietly failing because we pass in an empty string somewhere.